language/move-vm/runtime/src/interpreter.rs

```diff
@@ -262,18 +265,45 @@ impl Interpreter {
     ///
     /// Native functions do not push a frame at the moment and as such errors from a native
     /// function are incorrectly attributed to the caller.
-    fn make_call_frame(&mut self, func: Arc<Function>, ty_args: Vec<Type>) -> VMResult<Frame> {
+    fn make_call_frame(
+        &mut self,
+        func: Arc<Function>,
+        ty_args: Vec<Type>,
+        loader: &Loader,
+    ) -> VMResult<Frame> {
+        let resolver = func.get_resolver(loader);
         let mut locals = Locals::new(func.local_count());
         let arg_count = func.arg_count();
         for i in 0..arg_count {
+            let v = self.operand_stack.pop().map_err(|e| self.set_location(e))?;
+            if self.strict {
+                let expected_ty = resolver
+                    .resolve_signature_token(&func.local_types()[arg_count - i - 1])
```

Comment:
This is rather expensive. It could be made better performing by precomputing or caching things here. We can also keep for now, but I don't think we want to turn this on in every node then.

Reply:
We should be able to cache this part relatively easily.